core/src/main/java/org/apache/calcite/adapter/jdbc/JdbcCatalogSchema.java

```diff
@@ -80,7 +79,7 @@ public JdbcCatalogSchema(DataSource dataSource, SqlDialect dialect,
     this.dialect = requireNonNull(dialect, "dialect");
     this.convention = requireNonNull(convention, "convention");
     this.catalog = catalog;
-    this.subSchemas = new CachingLookup<>(new IgnoreCaseLookup<JdbcSchema>() {
```

Comment:
I would like to keep this. Otherwise, we will get at least 2 database queries for each table used within a query during preparation.